components/ui/shareIcon/index.tsx

```diff
@@ -0,0 +1,54 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
+
+import { postFeedItem } from "@/lib/feed";
+
+import Share from "./icon";
+
+import type { User } from "next-auth";
+
+const ShareIcon = ({
+  user,
+  sharedItem,
+}: {
+  user: User;
+  sharedItem:
+    | SpotifyApi.TrackObjectFull
+    | SpotifyApi.RecommendationTrackObject
+    | SpotifyApi.PlaylistObjectSimplified
+    | SpotifyApi.AlbumObjectSimplified
+    | SpotifyApi.ArtistObjectFull
+    | undefined;
+}) => {
+  const router = useRouter();
+  const [isPending, startTransition] = useTransition();
+  const [isFetching, setIsFetching] = useState(false);
+
+  // Create inline loading UI
+  const isMutating = isFetching || isPending;
+  const handleClick = async (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    setIsFetching(true);
+    await postFeedItem(user, "", sharedItem);
```

Comment:
We should let them add content, but I'm ok with doing that in a separate issue. Can you make one?

Maybe have the feed text box show the same compact card somehow? might require some refactoring...

Reply:
or clicking share pops up with a modal with a new instnatiation of FeedBox that takes in the item...

Reply:
Issue #255